src/solvers/sat/external_sat.cpp

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************\
+
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+\*******************************************************************/
+
+#include "external_sat.h"
+
+#include "dimacs_cnf.h"
+
+#include <util/exception_utils.h>
+#include <util/run.h>
+#include <util/string_utils.h>
+#include <util/tempfile.h>
+
+#include <json/json_parser.h>
+
+#include <chrono>
+#include <cstdlib>
+#include <fstream>
+#include <random>
+#include <sstream>
+#include <string>
+#include <thread>
+
+class missing_configurationt : public cprover_exception_baset
```

Comment:
This isn’t thrown anywhere right now. If I’m guessing the intent for where this is eventually supposed to be correctly this should probably be something like `invalid_command_line_argument_exceptiont`